code/code/cmd/cmd_deathstroke.cc

```diff
@@ -5,222 +5,168 @@
 
-  int dam = caster->getSkillDam(victim, SKILL_DEATHSTROKE, lev, caster->getAdvLearning(SKILL_DEATHSTROKE));
+  // Assessing an armor penalty regardless of success - this penalty greatly scales with level
+  // to ensure that players are not overly penalized early on
+  aff1.type = SKILL_DEATHSTROKE;
+  aff1.duration = Pulse::UPDATES_PER_MUDHOUR / 3;
+  aff1.location = APPLY_ARMOR;
+  aff1.modifier = (caster->GetMaxLevel()*caster->GetMaxLevel())/5 - skillLevel;
+  aff1.bitvector = 0;
 
-  if ((caster->bSuccess(bKnown+ percent, SKILL_DEATHSTROKE) &&
-         (i = caster->specialAttack(victim,SKILL_DEATHSTROKE)) && 
-         (i != GUARANTEED_FAILURE)) ||
-      !victim->awake()) {
+  // Success use case
+  if (!victim->awake() || 
+    (successfulHit && successfulSkill && successfulHit != GUARANTEED_FAILURE)) {
     wasSuccess = TRUE;
+
+    // Handling the use-case where getSkillDam() returns 0
     if (!dam) {
-      act("$n's attempt at $N's vital area falls far short of hitting.",
+      act("$n's attempt at $N's vital area is ineffective.",
                 FALSE, caster, 0, victim, TO_NOTVICT);
-      act("You fail to hit $N's vital area.", FALSE, caster, 0, victim, 
+      act("You hit to $N's vital area is ineffective.", FALSE, caster, 0, victim, 
                 TO_CHAR);
-      act("$n attempts to hit your vital area, but fails miserably.", 
+      act("$n attempts to hit your vital area, but the blow is ineffective.", 
                 FALSE, caster, 0, victim, TO_VICT);
+
     } else {
       act("$n hits $N in $S vital organs!", FALSE, caster, 0, victim, 
                 TO_NOTVICT);
       act("You hit $N in $S vital organs!", FALSE, caster, 0, victim, 
                 TO_CHAR);
       act("$n hits you in your vital organs!", FALSE, caster, 0, 
                 victim, TO_VICT);
+
+      // You successfully landed your deathstroke - assessing a bonus
+      aff2.type = SKILL_DEATHSTROKE;
+      aff2.duration = Pulse::UPDATES_PER_MUDHOUR / 3;
+      aff2.location = APPLY_HITROLL;
+      aff2.modifier = 3;
+      aff2.bitvector = 0;
     }
-    TThing *ob = caster->heldInPrimHand();
+
+    ob = caster->heldInPrimHand();
     if (ob && ob->isBluntWeapon())
       sktype = DAMAGE_CAVED_SKULL;
-    if (caster->reconcileDamage(victim, dam,sktype) == -1)
+    if (caster->reconcileDamage(victim, dam,sktype) == -1) 
       return DELETE_VICT;
-  } else {
+  }
+  // Failure use case
+  else {
     if (victim->getPosition() > POSITION_DEAD) {
       act("$n's attempt at $N's vital area falls far short of hitting.", 
                   FALSE, caster, 0, victim, TO_NOTVICT);
       act("You fail to hit $N's vital area.", FALSE, caster, 0, victim, 
                   TO_CHAR);
       act("$n attempts to hit your vital area, but fails miserably.", 
                   FALSE, caster, 0, victim, TO_VICT);
-      if (caster->reconcileDamage(victim, 0,sktype) == -1)
-        return DELETE_VICT;
     }
```

Comment:
else what? Can it happen? Print something?

Reply:
I think the else is it just continues through the function without returning.  It goes on to allow the victim a chance to counter attack down below and ultimately will return an appropriate value, but there's not specifically an else case to handle here I believe.